interfaces/cython/cantera/_onedim.pyx

```diff
@@ -10,15 +10,18 @@ from ._utils cimport stringify, pystr
 from ._utils import CanteraError
 from cython.operator import dereference as deref
 
+
 # Need a pure-python class to store weakrefs to
 class _WeakrefProxy:
     pass
 
 cdef class Domain1D:
-    def __cinit__(self, *args, **kwargs):
+    def __cinit__(self, _SolutionBase phase, **kwargs):
+        if not isinstance(phase, _SolutionBase):
+            raise TypeError(f"Received phase with invalid type '{type(phase)}'.")
```

Comment:
This should be sufficient to raise a `TypeError` when needed:
```suggestion
    def __cinit__(self, _SolutionBase phase not None, **kwargs):
```

Reply:
nice - didn't know this one!